--- a/include/class.dynamic_forms.php
+++ b/include/class.dynamic_forms.php
@@ -300,8 +300,13 @@ class TicketForm extends DynamicForm {
         // ans.entry_id = entry.id LEFT JOIN ost_form_field field ON
         // field.id=ans.field_id
         // where entry.object_type='T' group by entry.object_id;
-        $sql = 'CREATE TABLE `'.TABLE_PREFIX.'ticket__cdata` (PRIMARY KEY
-                (ticket_id)) AS ' . static::getCrossTabQuery('T', 'ticket_id');
+        // $sql = 'CREATE TABLE `'.TABLE_PREFIX.'ticket__cdata` (PRIMARY KEY
+        //        (ticket_id)) AS ' . static::getCrossTabQuery('T', 'ticket_id');
+        // This will cause GTID violation; see 
+        // https://github.com/osTicket/osTicket/issues/3384
+        // Use view instead
+        $sql = 'CREATE VIEW `'.TABLE_PREFIX.'ticket__cdata` AS ' 
+                . static::getCrossTabQuery('T', 'ticket_id');
         db_query($sql);
     }
 