--- a/include/class.dynamic_forms.php
+++ b/include/class.dynamic_forms.php
@@ -286,9 +286,8 @@
     }

     static function buildDynamicDataView($cdata) {
-        $sql = 'CREATE TABLE IF NOT EXISTS `'.$cdata['table'].'` (PRIMARY KEY
-                ('.$cdata['object_id'].')) DEFAULT CHARSET=utf8 AS '
-             .  static::getCrossTabQuery( $cdata['object_type'], $cdata['object_id']);
+        $sql = 'CREATE VIEW `'.TABLE_PREFIX.'ticket__cdata` AS '
+                . static::getCrossTabQuery('T', 'ticket_id');
         db_query($sql);
     }
